packages/plugin-types/API.md

```diff
@@ -39,6 +40,13 @@ export type AuthResources = {
     authenticatedUserIamRole: IRole;
     unauthenticatedUserIamRole: IRole;
     cfnResources: AuthCfnResources;
+    groups: {
+        [key: string]: {
+            groupName: string;

```

Comment:
If the key is also `groupName` do we need this in object as well?

Reply:
This is just for clarity, [key: string] doesn't support JS doc so it isn't known unless you are looking at the docs: 
<img width="274" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/110861985/6c6ca111-9a78-42bf-91a3-57fbc8790d7c">

Reply:
ie:
<img width="323" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/110861985/bf05fa64-b6f8-4604-8495-6cb5003a8af5">

Reply:
Could we do `[groupName: string]` as the key definition?

Reply:
Just to refer/destructure this type, customers "have to" know that the key is `groupName`, right?

Reply:
We could also add JSDocs on the group field itself explaining that the value is a record that maps group name to corresponding roles and userpool group

Reply:
Updating to this, and removing the 'groupName' property. Thanks for the suggestions, 
<img width="630" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/110861985/7c20856e-9403-4687-af98-c04ded4d90df">